mozilla/fx-private-relay
# MPP-2802 - Update premium onboarding, subdomain picker and confirmation modal

# New feature description

Acceptance criteria: https://mozilla-hub.atlassian.net/jira/software/c/projects/MPP/boards/198?modal=detail&selectedIssue=MPP-2802&assignee=60d0cb9820d2dd00702f8271
Dependency: https://github.com/mozilla-l10n/fx-private-relay-l10n/pull/137

This updates the premium onboarding UI, subdomain picker banner and subdomain confirmation modals.
- [x] Onboarding step 1: welcome
- [x] Onboarding step 2: set subdomain
- [x] Onboarding step 3: add extension
- [x] Confirmation warning modal
- [x] Confirmation success modal
- [x] Subdomain banner

# Screenshot (if applicable)

<img width="999" alt="image" src="https://user-images.githubusercontent.com/13066134/226463301-8ea649e0-d113-4da4-9340-c71f91adc490.png">
<img width="991" alt="image" src="https://user-images.githubusercontent.com/13066134/226463334-29c74114-dadf-4d17-bea4-dd502af155fb.png">
<img width="983" alt="image" src="https://user-images.githubusercontent.com/13066134/226463353-f677f429-0049-46aa-a501-c47e4e19050e.png">
<img width="711" alt="image" src="https://user-images.githubusercontent.com/13066134/226463371-8c1828de-5021-40ca-97fa-cfaf13419054.png">
<img width="761" alt="image" src="https://user-images.githubusercontent.com/13066134/226463393-0f224650-2a73-4450-b356-fa6a962c28b6.png">


# How to test

Onboarding: Set your onboarding state to 0 and subdomain to null in the admin settings of a premium account, walk through the onboarding process ensure that the flow follows that of the screenshots above.
Custom domain banner: Remove the subdomain again and on the profile, check that the subdomain looks and functions as intended.

# Checklist (Definition of Done)
